pallets/pallet-deposit-storage/src/lib.rs

```diff
@@ -115,6 +118,7 @@ pub mod pallet {
 	#[pallet::composite_enum]
 	pub enum HoldReason {
 		Deposit,
+		FungibleImpl,
```

Comment:
I think a better name would be `SystemDeposit`

Reply:
Mhh ok, but what if the pallet will implement a new trait, and we need a new deposit reason? In that case, it would need to be different than this one, while it could still be stored within the same storage map.

Reply:
Okay, maybe I am missing something. I was thinking the `FungibleImpl` reason is introduced to separate regular and system deposits from each other.

Can you perhaps explain what you mean with 'implementing a new trait'?

Reply:
`FungibleImpl`, in my opinion, is to "namespace" deposits taken when implementing the `MutateHold` trait. If we were to implement, let's say, the `fungibles::MutateHold` trait tomorrow, I would rather see a new hold reason `FungiblesImpl` which we would use when implementing that trait. Does it make sense?

Reply:
After all, that's what hold reasons are for, right? So that one implementation does not interfere (i.e., cannot release) deposits taken with a different hold reason, i.e., with a different trait implementation.

Reply:
i hope this clarifies better the use of the `HoldReason` for this pallet?

Reply:
Okay, I think I understand. Fine for me :+1: